src/extensions/default/Inline3DParametersEditor/Parameters3DUtils.js

```diff
@@ -0,0 +1,44 @@
+/**
+ *  Utilities functions related to 3-D parameter matching
+ *
+ */
+define(function (require, exports, module) {
+    "use strict";
+
+    var PARAMETERS_3D_REGEX = /( )*(((-)?\d+(\.\d+)?)( )+)*((-)?\d+(\.\d+)?( )*)/g;
```

Comment:
When you use regex, *document them* because regex are incredibly complicated things and can even be used to self-DOS a codebase if you're not careful. So, looking at this on https://www.debuggex.com/r/ecGjHVcRHihRQzed it seems you're doing the following:
1. skip over initial spaces, but for some weird reason only *capture* a single space. (you're using `( )*` rather than `( *)`, which should really be `(\s*)`, although I don't understand why you're capturing spaces at all here)
2. match a number that can be negative, and can be decimal, but **must** be followed by a space. Do this one or more times.
3. match the same kind of number again, but this time followed by an *optional* space.

This is not a good pattern to use for matching. If you need to match 3D parameters, then simply use this:
```
  // group 1: repetition group
  // group 2: (possibly negative) integer number
  // group 3: decimal fraction
  var pattern = / (\s?(-?\d+(\.\d+)*))+/g
```